### ISSUE TITLE
flynn cluster backup --file backup.tar gets stuck

### ISSUE THREAD

**Issue by joeczucha** [flynn/flynn#4387]
Running the above command gets stuck at the same point every time:

    $ flynn cluster backup --file backup.tar
    Creating cluster backup...
    6.35 MB 810.58 KB/s

I left it for ~15 minutes and no movement.

I've tried running it on both my local and on the host itself, same result. Then tried running `flynn-host fix` (which completes without error) but the same thing happens afterwards. 

    $ flynn-host fix -n 3
    INFO[05-17|12:04:09] found expected hosts                     n=3
    INFO[05-17|12:04:09] ensuring discoverd is running on all hosts
    INFO[05-17|12:04:09] checking flannel
    INFO[05-17|12:04:09] flannel looks good
    INFO[05-17|12:04:09] waiting for discoverd to be available
    INFO[05-17|12:04:09] checking for running controller API
    INFO[05-17|12:04:09] found running controller API instances   n=2
    INFO[05-17|12:04:09] found controller instance, checking critical formations
    INFO[05-17|12:04:09] checking status of sirenia databases
    INFO[05-17|12:04:09] checking for database state              db=postgres
    INFO[05-17|12:04:09] checking sirenia cluster status          fn=CheckSirenia service=postgres
    INFO[05-17|12:04:09] found running leader                     fn=CheckSirenia service=postgres
    INFO[05-17|12:04:09] found running instances                  fn=CheckSirenia service=postgres count=3
    INFO[05-17|12:04:09] getting sirenia status                   fn=CheckSirenia service=postgres
    INFO[05-17|12:04:09] cluster claims to be read-write          fn=CheckSirenia service=postgres
    INFO[05-17|12:04:09] checking for database state              db=mariadb
    INFO[05-17|12:04:09] checking sirenia cluster status          fn=CheckSirenia service=mariadb
    INFO[05-17|12:04:09] found running leader                     fn=CheckSirenia service=mariadb
    INFO[05-17|12:04:09] found running instances                  fn=CheckSirenia service=mariadb count=3
    INFO[05-17|12:04:09] getting sirenia status                   fn=CheckSirenia service=mariadb
    INFO[05-17|12:04:10] cluster claims to be read-write          fn=CheckSirenia service=mariadb
    INFO[05-17|12:04:10] checking for database state              db=mongodb
    INFO[05-17|12:04:10] skipping recovery of db, no state in discoverd db=mongodb
    INFO[05-17|12:04:10] checking for running controller API
    INFO[05-17|12:04:10] found running controller API instances   n=2
    INFO[05-17|12:04:10] found controller instance, checking critical formations
    INFO[05-17|12:04:10] scheduler looks up, moving on
    INFO[05-17|12:04:10] cluster fix complete

Any other tips on where to start troubleshooting this? 

Thanks

**Comment by lmars** [comment]
Send the process a SIGQUIT signal (hit <kbd>ctrl + \\</kbd>) when it's stuck and post the resulting stack trace.

**Comment by joeczucha** [comment]
Thanks @lmars 

    ^\SIGQUIT: quit
    PC=0x10589cb m=0 sigcode=0

    goroutine 0 [idle]:
    runtime.mach_semaphore_wait(0x803, 0x10201ed, 0x16ac220, 0x0, 0xc42049c800, 0x16abee0, 0x7ffeefbff750, 0x1052fa3, 0xffffffffffffffff, 0xc42049c800, ...)
      /usr/local/go/src/runtime/sys_darwin_amd64.s:445 +0xb
    runtime.semasleep1(0xffffffffffffffff, 0xc42049c800)
      /usr/local/go/src/runtime/os_darwin.go:413 +0x52
    runtime.semasleep.func1()
      /usr/local/go/src/runtime/os_darwin.go:432 +0x33
    runtime.systemstack(0x7ffeefbff778)
      /usr/local/go/src/runtime/asm_amd64.s:360 +0xab
    runtime.semasleep(0xffffffffffffffff, 0xc420021860)
      /usr/local/go/src/runtime/os_darwin.go:431 +0x44
    runtime.notesleep(0x16ac730)
      /usr/local/go/src/runtime/lock_sema.go:167 +0xe9
    runtime.stopm()
      /usr/local/go/src/runtime/proc.go:1680 +0xe5
    runtime.findrunnable(0xc420020600, 0x0)
      /usr/local/go/src/runtime/proc.go:2135 +0x4d2
    runtime.schedule()
      /usr/local/go/src/runtime/proc.go:2255 +0x12c
    runtime.park_m(0xc42022f080)
      /usr/local/go/src/runtime/proc.go:2318 +0xb6
    runtime.mcall(0x7ffeefbff960)
      /usr/local/go/src/runtime/asm_amd64.s:286 +0x5b

    goroutine 1 [IO wait, 85 minutes]:
    internal/poll.runtime_pollWait(0x17c9df0, 0x72, 0x0)
      /usr/local/go/src/runtime/netpoll.go:173 +0x57
    internal/poll.(*pollDesc).wait(0xc4202ca898, 0x72, 0xffffffffffffff00, 0x16729a0, 0x166e2d8)
      /usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0xae
    internal/poll.(*pollDesc).waitRead(0xc4202ca898, 0xc42036e800, 0x400, 0x400)
      /usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
    internal/poll.(*FD).Read(0xc4202ca880, 0xc42036e800, 0x400, 0x400, 0x0, 0x0, 0x0)
      /usr/local/go/src/internal/poll/fd_unix.go:126 +0x18a
    net.(*netFD).Read(0xc4202ca880, 0xc42036e800, 0x400, 0x400, 0xc42007bf20, 0xc4205f92f8, 0x10f08d4)
      /usr/local/go/src/net/fd_unix.go:202 +0x52
    net.(*conn).Read(0xc420272018, 0xc42036e800, 0x400, 0x400, 0x0, 0x0, 0x0)
      /usr/local/go/src/net/net.go:176 +0x6d
    crypto/tls.(*block).readFromUntil(0xc42017a2a0, 0x17ce2a8, 0xc420272018, 0x5, 0xc420272018, 0xc4205f9318)
      /usr/local/go/src/crypto/tls/conn.go:488 +0x95
    crypto/tls.(*Conn).readRecord(0xc420033180, 0x148c517, 0xc4200332a0, 0xc4202c5000)
      /usr/local/go/src/crypto/tls/conn.go:590 +0xe0
    crypto/tls.(*Conn).Read(0xc420033180, 0xc4202c5000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
      /usr/local/go/src/crypto/tls/conn.go:1134 +0x110
    bufio.(*Reader).fill(0xc42005a540)
      /usr/local/go/src/bufio/bufio.go:97 +0x11a
    bufio.(*Reader).ReadByte(0xc42005a540, 0xc420014798, 0x1000, 0xc4202c5000)
      /usr/local/go/src/bufio/bufio.go:239 +0x39
    github.com/flynn/flynn/pkg/cluster.(*attachClient).Receive(0xc42007bf50, 0x1670460, 0xc4202326c0, 0x1670de0, 0xc42000c020, 0x1000, 0x0, 0x0)
      /go/src/github.com/flynn/flynn/pkg/cluster/attach.go:154 +0xd9
    github.com/flynn/flynn/pkg/backup.(*TarWriter).runJob(0xc420238120, 0x167e3a0, 0xc420272000, 0x146c20e, 0x7, 0xc42026e1e0, 0x1670460, 0xc4202326c0, 0x0, 0x0)
      /go/src/github.com/flynn/flynn/pkg/backup/tar_writer.go:122 +0x27b
    github.com/flynn/flynn/pkg/backup.(*TarWriter).WriteCommandOutput(0xc420238120, 0x167e3a0, 0xc420272000, 0x146df41, 0xc, 0x146c20e, 0x7, 0xc42026e1e0, 0x0, 0x0)
      /go/src/github.com/flynn/flynn/pkg/backup/tar_writer.go:86 +0x215
    github.com/flynn/flynn/pkg/backup.Run(0x167e3a0, 0xc420272000, 0x1670de0, 0xc420272008, 0x16736a0, 0xc42026e0f0, 0x0, 0x0)
      /go/src/github.com/flynn/flynn/pkg/backup/backup.go:57 +0x115b
    main.runClusterBackup(0xc4202137c0, 0x0, 0x0)
      /go/src/github.com/flynn/flynn/cli/cluster.go:430 +0x1cc
    main.runCluster(0xc4202137c0, 0x0, 0x0)
      /go/src/github.com/flynn/flynn/cli/cluster.go:123 +0x66f
    main.runCommand(0x7ffeefbffb39, 0x7, 0xc420056600, 0x3, 0x4, 0xc117b801, 0x14c2170)
      /go/src/github.com/flynn/flynn/cli/main.go:178 +0x316
    main.main()
      /go/src/github.com/flynn/flynn/cli/main.go:128 +0x659

    goroutine 5 [syscall, 85 minutes]:
    os/signal.signal_recv(0x0)
      /usr/local/go/src/runtime/sigqueue.go:131 +0xa7
    os/signal.loop()
      /usr/local/go/src/os/signal/signal_unix.go:22 +0x22
    created by os/signal.init.0
      /usr/local/go/src/os/signal/signal_unix.go:28 +0x41

    goroutine 6 [chan receive, 85 minutes]:
    github.com/flynn/flynn/pkg/shutdown.(*handler).wait(0xc42007b290)
      /go/src/github.com/flynn/flynn/pkg/shutdown/handler.go:58 +0xd4
    created by github.com/flynn/flynn/pkg/shutdown.newHandler
      /go/src/github.com/flynn/flynn/pkg/shutdown/handler.go:25 +0x88

    goroutine 17 [select, 85 minutes, locked to thread]:
    runtime.gopark(0x148c4b0, 0x0, 0x146bdac, 0x6, 0x18, 0x1)
      /usr/local/go/src/runtime/proc.go:287 +0x12c
    runtime.selectgo(0xc420044f50, 0xc420110060)
      /usr/local/go/src/runtime/select.go:395 +0x1149
    runtime.ensureSigM.func1()
      /usr/local/go/src/runtime/signal_unix.go:511 +0x1fe
    runtime.goexit()
      /usr/local/go/src/runtime/asm_amd64.s:2337 +0x1

    goroutine 33 [sleep]:
    time.Sleep(0xbebc200)
      /usr/local/go/src/runtime/time.go:65 +0x130
    github.com/flynn/flynn/vendor/github.com/cheggaaa/pb.(*ProgressBar).writer(0xc42026e0f0)
      /go/src/github.com/flynn/flynn/vendor/github.com/cheggaaa/pb/pb.go:332 +0x3a
    created by github.com/flynn/flynn/vendor/github.com/cheggaaa/pb.(*ProgressBar).Start
      /go/src/github.com/flynn/flynn/vendor/github.com/cheggaaa/pb/pb.go:101 +0x97

    goroutine 39 [IO wait, 85 minutes]:
    internal/poll.runtime_pollWait(0x17c9eb0, 0x72, 0x0)
      /usr/local/go/src/runtime/netpoll.go:173 +0x57
    internal/poll.(*pollDesc).wait(0xc420292518, 0x72, 0xffffffffffffff00, 0x16729a0, 0x166e2d8)
      /usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0xae
    internal/poll.(*pollDesc).waitRead(0xc420292518, 0xc42030a000, 0x1000, 0x1000)
      /usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
    internal/poll.(*FD).Read(0xc420292500, 0xc42030a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
      /usr/local/go/src/internal/poll/fd_unix.go:126 +0x18a
    net.(*netFD).Read(0xc420292500, 0xc42030a000, 0x1000, 0x1000, 0x1, 0x3, 0xc420049788)
      /usr/local/go/src/net/fd_unix.go:202 +0x52
    net.(*conn).Read(0xc42000cf70, 0xc42030a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
      /usr/local/go/src/net/net.go:176 +0x6d
    crypto/tls.(*block).readFromUntil(0xc42017b230, 0x17ce2a8, 0xc42000cf70, 0x5, 0xc42000cf70, 0xc4202d9320)
      /usr/local/go/src/crypto/tls/conn.go:488 +0x95
    crypto/tls.(*Conn).readRecord(0xc4201dee00, 0x148c517, 0xc4201def20, 0x102b8eb)
      /usr/local/go/src/crypto/tls/conn.go:590 +0xe0
    crypto/tls.(*Conn).Read(0xc4201dee00, 0xc4202ad000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
      /usr/local/go/src/crypto/tls/conn.go:1134 +0x110
    net/http.(*persistConn).Read(0xc42029e000, 0xc4202ad000, 0x1000, 0x1000, 0xc42026c150, 0xc420078658, 0x1051aa0)
      /usr/local/go/src/net/http/transport.go:1391 +0x140
    bufio.(*Reader).fill(0xc4201266c0)
      /usr/local/go/src/bufio/bufio.go:97 +0x11a
    bufio.(*Reader).Peek(0xc4201266c0, 0x1, 0x0, 0x0, 0x0, 0xc4202f6780, 0x0)
      /usr/local/go/src/bufio/bufio.go:129 +0x3a
    net/http.(*persistConn).readLoop(0xc42029e000)
      /usr/local/go/src/net/http/transport.go:1539 +0x185
    created by net/http.(*Transport).dialConn
      /usr/local/go/src/net/http/transport.go:1186 +0xa2e

    goroutine 40 [select, 85 minutes]:
    net/http.(*persistConn).writeLoop(0xc42029e000)
      /usr/local/go/src/net/http/transport.go:1759 +0x165
    created by net/http.(*Transport).dialConn
      /usr/local/go/src/net/http/transport.go:1187 +0xa53

    rax    0xe
    rbx    0x16ac620
    rcx    0x7ffeefbff6f0
    rdx    0x7ffeefbff778
    rdi    0x803
    rsi    0xc42049c8000201
    rbp    0x7ffeefbff728
    rsp    0x7ffeefbff6f0
    r8     0xa
    r9     0x16abee0
    r10    0x40
    r11    0x286
    r12    0x13aa85bd014c6
    r13    0x0
    r14    0x10562d0
    r15    0x0
    rip    0x10589cb
    rflags 0x286
    cs     0x7
    fs     0x0
    gs     0x0

**Comment by lmars** [comment]
@joeczucha ok so it is blocked pulling the MySQL dump, if you run `flynn-host ps` on a host in your cluster do you see any failed MySQL jobs?

**Comment by joeczucha** [comment]
I think you might be on to something there :)

    > show processlist;
    +-------+----------------------------------+----------------------+----------------------------------+---------+------+--------------------------------------+------------------------------------------------------------------------------------------------------+----------+
    | Id    | User                             | Host                 | db                               | Command | Time | State                                | Info                                                                                                 | Progress |
    +-------+----------------------------------+----------------------+----------------------------------+---------+------+--------------------------------------+------------------------------------------------------------------------------------------------------+----------+
    | 39587 | flynn                            | 100.100.48.91:57434  | mysql                            | Sleep   |  350 |                                      | NULL                                                                                                 |    0.000 |
    | 39588 | flynn                            | 100.100.28.0:52248   | mysql                            | Sleep   |  350 |                                      | NULL                                                                                                 |    0.000 |
    | 39593 | flynn                            | 100.100.28.0:52712   | mysql                            | Sleep   |  350 |                                      | NULL                                                                                                 |    0.000 |
    | 39594 | flynn                            | 100.100.48.92:55972  | mysql                            | Sleep   |  350 |                                      | NULL                                                                                                 |    0.000 |
    | 39595 | flynn                            | 100.100.28.0:43794   | mysql                            | Sleep   |  350 |                                      | NULL                                                                                                 |    0.000 |
    | 39596 | flynn                            | 100.100.48.110:59038 | mysql                            | Sleep   |  350 |                                      | NULL                                                                                                 |    0.000 |
    | 39600 | flynn                            | 100.100.83.0:55472   | mysql                            | Sleep   |  350 |                                      | NULL                                                                                                 |    0.000 |
    | 39608 | flynn                            | 100.100.83.0:42694   | mysql                            | Sleep   |  350 |                                      | NULL                                                                                                 |    0.000 |
    | 39620 | flynn                            | 100.100.28.0:42612   | e65e6c6dab9536540e339a99ca3bc92d | Query   |    0 | init                                 | show processlist                                                                                     |    0.000 |
    | 39623 | flynn                            | 100.100.83.0:47516   | mysql                            | Sleep   |  110 |                                      | NULL                                                                                                 |    0.000 |
    | 39637 | 4566e719cbb88cd350b82337b940d5fa | 100.100.83.0:51148   | e65e6c6dab9536540e339a99ca3bc92d | Query   |   13 | Waiting for semi-sync ACK from slave | UPDATE `lf_options` SET `option_value` = '1526591052.1528201103210449218750' WHERE `option_name` = ' |    0.000 |
    | 39638 | 4566e719cbb88cd350b82337b940d5fa | 100.100.83.0:51150   | e65e6c6dab9536540e339a99ca3bc92d | Query   |   13 | Waiting for table level lock         | SELECT option_name, option_value FROM lf_options WHERE autoload = 'yes'                              |    0.000 |
    | 39639 | 4566e719cbb88cd350b82337b940d5fa | 100.100.83.0:51152   | e65e6c6dab9536540e339a99ca3bc92d | Query   |   13 | Waiting for table level lock         | SELECT option_name, option_value FROM lf_options WHERE autoload = 'yes'                              |    0.000 |
    | 39641 | 4566e719cbb88cd350b82337b940d5fa | 100.100.83.0:51212   | e65e6c6dab9536540e339a99ca3bc92d | Query   |    0 | Waiting for table level lock         | SELECT option_name, option_value FROM lf_options WHERE autoload = 'yes'                              |    0.000 |
    +-------+----------------------------------+----------------------+----------------------------------+---------+------+--------------------------------------+------------------------------------------------------------------------------------------------------+----------+

    $ flynn -a mariadb ps
    ID                                            TYPE     STATE  CREATED        RELEASE
    flynn02-d51f5a18-862a-452a-b065-a404d41b349b  mariadb  up     8 weeks ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn02-c68f0d8a-7b3a-4860-befa-dd3ff5ade52f  web      up     8 weeks ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn03-11912440-46f6-4a71-9d08-8b70a4095a77  web      up     8 weeks ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn03-b14af6e7-f0fa-4ba9-858f-f45c13881851  mariadb  up     8 weeks ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn03-aa3c307d-1d02-46c1-b02b-73d5cdaaa02e  run      up     9 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn02-7d9bf024-7f18-4e63-94e1-72fa0a232d37  run      up     9 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn02-c83ce6e2-ecbb-42d9-a968-cf9210d51752  run      up     9 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn03-2dca90fa-b116-427f-8484-532bc8007844  run      up     9 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn02-02040515-6c09-44d6-a295-85173e75dfac  run      up     9 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn02-37da029d-dcb5-42ad-be56-7566365f5ad5  mariadb  up     9 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn03-4282d1c9-9c7b-4367-b9d2-28aa397c81b1  run      up     9 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn01-694c15ec-9b69-4041-b2b2-5a1e8ed261b0  run      up     9 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn01-8a7c7722-3f02-4d39-8b44-d8676848f867  run      up     6 hours ago    1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn02-901b311b-0a94-42a2-94bc-392b10afe4de  run      up     9 minutes ago  1ad0283b-551a-49a9-a105-068a5c6796f9
    flynn01-4d1696dd-b6d8-4ef9-bd17-e488611fe28b  run      up     5 minutes ago  1ad0283b-551a-49a9-a105-068a5c6796f9

I tried killing the processes with the `Waiting for...` statuses but more just reappear... would a reboot help?

**Comment by philiplb** [comment]
I wonder if this is similar to https://github.com/flynn/flynn/issues/4144
Does the MariaDB report healthy for you on the dashboard?

**Comment by joeczucha** [comment]
@philiplb Strangely enough the mariadb is showing as healthy but the mongodb has a red line under it. I don't use mongodb for anything, though...

**Comment by joeczucha** [comment]
I've managed to fix this so will post my solution here for posterity should anyone face the same issue and want something to try. Of course, YMMV.

So, as per the above, I tried killing the rogue MYSQL processes in the hope that it would unlock the tables and the backup would complete. No dice - as I killed processes more just appeared with the "Waiting for..." message. However, in doing this, I also managed to unwittingly break the site; visiting it was now giving me a 503 error.

Next I tried restarting all 3 boxes in my cluster. The 503 disappeared but was replaced by a "Service unavailable" message, even though it seemed to have the web process running just fine. I tried running `flynn scale web=0` and then `flynn scale web=1` to see if that would make a difference, but I just kept getting the same "Service unavailable" message*.

Finally, I redeployed my app using `git push`. Following successful deployment, the app came back to life and the locking jobs have disappeared from MYSQL. 

I'm still a bit nervous about it's stability so, when I have more time, I'll probably do a cluster rebuild but, for now, at least the backups are completing!

Thanks again for your help. I'll close this issue tomorrow but I'm also happy send any logs that you think may be useful.

* I'm not entirely sure whether it was the redeployment or the `flynn scale` commands that did the trick... upon redeployment I tried a different browser and the site worked so I emptied my Chrome cache and, at that point, it started working too.
